view/src/components/produce.js

```diff
@@ -0,0 +1,713 @@
+                          </Typography>
+                          <Typography variant="body2" component="p">
+                            USDA Price: ${produce.price}
+                            <br />
+                            Average Price Paid by Farmlink: ${produce.pricePaid}
+                          </Typography>
+                        </Box>
+                        <Box p={3}>
+                          <Typography
+                            className={classes.pos}
+                            color="textSecondary"
+                          >
+                            Internal Statistics:
+                          </Typography>
+                          <Typography variant="body2" component="p">
+                            Amount Moved by FarmLink (lbs):{" "}
+                            {produce.amountMoved}
+                          </Typography>
+                        </Box>
+                      </Box>
+                    </CardContent>
+                    <CardActions>
+                      <Button
+                        size="small"
+                        color="primary"
+                        onClick={() => this.handleViewOpen({ produce })}
+                      >
+                        {" "}
+                        View{" "}
+                      </Button>
+                      <Button
+                        size="small"
+                        color="primary"
+                        onClick={() => this.handleEditClickOpen({ produce })}
+                      >
+                        Edit
+                      </Button>
+                      <Button
+                        size="small"
+                        color="primary"
+                        onClick={() => this.deleteTodoHandler({ produce })}
+                      >
+                        Delete
+                      </Button>
+                    </CardActions>
+                  </Card>
+                </Grid>
+              ))}
+            </Grid>
+          </Container>
+
+          <Dialog
+            onClose={handleViewClose}
+            aria-labelledby="customized-dialog-title"
+            open={viewOpen}
+            fullWidth
+            classes={{ paperFullWidth: classes.dialogStyle }}
+          >
+            <DialogTitle id="customized-dialog-title" onClose={handleViewClose}>
+              Stone Fruits
+              {this.state.name}
+            </DialogTitle>
+            <DialogContent dividers>
+              <Box
+                display="flex"
+                flexDirection="row"
+                flexWrap="wrap"
+                p={0}
+                m={0}
+              >
+                <Box p={3}>
+                  <Typography className={classes.pos} color="textSecondary">
+                    Shipping Temperatures in Reefer (°F):
+                  </Typography>
+                  <Typography variant="body2" component="p">
+                    Maintenance Temperature:{" "}
```

Comment:
I have a theory about these.  The way HTML works is that any contiguous run of whitespace characters in the source text gets converted to a single space character when rendering.  A newline is a whitespace character.  So, surprisingly:

```
<div>
  <span>Foo</span>
  </span>Bar</span>
</div>
```

renders as "Foo Bar", but:

```
<div><span>Foo</span></span>Bar</span></div>
```

renders as "FooBar".  I think prettier might be trying to reformat your code while still keeping the rendering the same, but either it's getting confused or the original rendering wasn't what you wanted anyway.  In this case, for example, the newline after "Maintenance Temperature:" ought to insert a space into the output even without this...  assuming that the JSX maps to HTML in a reasonable way without compressing away the whitespace.

Reply:
Intersting!